src/modules/peek/Peek.FilePreviewer/Previewers/UnsupportedFilePreviewer/UnsupportedFilePreviewer.cs

```diff
@@ -97,139 +101,59 @@ await Dispatcher.RunOnUiThread(async () =>
-                });
-            });
-
-            return isIconValid && isTaskSuccessful;
+            Preview.IconPreview = await ThumbnailHelper.GetThumbnailAsync(Item.Path, cancellationToken) ??
+                await ThumbnailHelper.GetIconAsync(Item.Path, cancellationToken) ??
+                DefaultIcon;
         }
 
-        public async Task<bool> LoadDisplayInfoAsync(CancellationToken cancellationToken)
+        private async Task LoadDisplayInfoAsync(IProgress<string> sizeProgress, CancellationToken cancellationToken)
         {
-            bool isDisplayValid = false;
-
-            var isTaskSuccessful = await TaskExtension.RunSafe(async () =>
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                var type = await Task.Run(Item.GetContentTypeAsync);
-
-                cancellationToken.ThrowIfCancellationRequested();
-
-                isDisplayValid = type != null;
-
-                var readableFileSize = string.Empty;
+            string type = await Item.GetContentTypeAsync();
 
-                if (Item is FileItem)
-                {
-                    readableFileSize = ReadableStringHelper.BytesToReadableString(Item.FileSizeBytes);
-                }
-                else if (Item is FolderItem)
-                {
-                    ComputeFolderSize(cancellationToken);
-                }
+            Dispatcher.TryEnqueue(() => Preview.FileType = type);
 
-                await Dispatcher.RunOnUiThread(() =>
-                {
-                    Preview.FileSize = readableFileSize;
-                    Preview.FileType = type;
-                    return Task.CompletedTask;
-                });
-            });
-
-            return isDisplayValid && isTaskSuccessful;
+            if (Item is FolderItem folderItem)
+            {
+                await Task.Run(() => CalculateFolderSizeWithProgress(Item.Path, sizeProgress, cancellationToken), cancellationToken);
+            }
+            else
+            {
+                ReportProgress(sizeProgress, Item.FileSizeBytes);
+            }
         }
 
-        private bool HasFailedLoadingPreview()
+        private void CalculateFolderSizeWithProgress(string path, IProgress<string> progress, CancellationToken cancellationToken)
         {
-            var isLoadingIconPreviewSuccessful = IconPreviewTask?.Result ?? false;
-            var isLoadingDisplayInfoSuccessful = DisplayInfoTask?.Result ?? false;
+            ulong folderSize = 0;
+            TimeSpan updateInterval = TimeSpan.FromMilliseconds(1000 / MaxUpdateFps);
+            DateTime nextUpdate = DateTime.UtcNow + updateInterval;
 
-            return !isLoadingIconPreviewSuccessful || !isLoadingDisplayInfoSuccessful;
-        }
+            var files = new DirectoryInfo(path).EnumerateFiles("*", new EnumerationOptions { RecurseSubdirectories = true });
```

Comment:
Let's move `new EnumerationOptions { RecurseSubdirectories = true }` in a static field to reuse it across multiple previews, although I don't think the overhead will be that high.

Reply:
Good point! I should have replicated that pattern from your original code.